### DISCUSSION TITLE
Microsoft Open JDK 17 and Kubernetes VPA

### DISCUSSION THREAD

**Discussion by dkirrane** [microsoft/openjdk#574] [Q&A] [answered]
For a containerized Java 17 application is it possible to use Kubernetes VPA to set container cpu & memory requests and limits?

I was thinking it should be possible if I set `-XX:MaxRAMPercentage=70` instead of fixing with `Xms` and `Xmx`.

Appreciate any other advice/best practices for JVM settings and GC to use for containers?

**Comment by karianna** [comment, marked as answer]
Yes you can, please see https://learn.microsoft.com/en-us/azure/developer/java/containers/overview?bc=%2Fazure%2Fspring-cloud%2Fbreadcrumb%2Ftoc.json for guidance.
